utils/messageqcpp/compressed_iss.h

```diff
@@ -54,7 +54,7 @@ class CompressedInetStreamSocket : public InetStreamSocket
     virtual const IOSocket accept(const struct timespec* timeout);
     virtual void connect(const sockaddr* addr);
 private:
-    compress::IDBCompressInterface alg;
+    std::shared_ptr<compress::CompressInterface> alg;
```

Comment:
Same question about `unique_ptr`

Reply:
has default copy cosntructor https://github.com/mariadb-corporation/mariadb-columnstore-engine/blob/develop/utils/messageqcpp/compressed_iss.h#L45 , but I'm not an expert in smart pointers, just seems to me `unique_ptr` does not have copy constructor, please fix me if I'm wrong.